### PR TITLE
nmap: use Python3 and add Nping with SSL option

### PR DESCRIPTION
**Pull request by nunojpg**

### CODE REVIEW

**Comment by neheb** [comment]
Any reason why fno-builtin is passed? Looks like some bugfix for uClibc++ or an old version of GCC.

**Comment by nunojpg** [comment]
i should test to remove it. Thanks.

File: net/nmap/Makefile [hunk 1]
```diff
@@ -13,13 +13,13 @@
...
-PKG_VERSION:=7.70
+PKG_VERSION:=7.80
 PKG_RELEASE:=1
 PKG_MAINTAINER:=Nuno Goncalves <nunojpg@gmail.com>
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=https://nmap.org/dist/
-PKG_HASH:=847b068955f792f4cc247593aca6dc3dc4aae12976169873247488de147a6e18
+PKG_HASH:=fcfa5a0e42099e12e4bf7a68ebe6fde05553383a682e816a7ec9256ab4773faa
 PKG_LICENSE:=GPL-2.0
```
Comment (neheb) [inline net/nmap/Makefile]: This should be GPL-2.0-only

**Comment by neheb** [comment]
PKG_BUILD_DIR can be removed. It matches the default value.

**Comment by neheb** [comment]
PKG_BUILD_PARALLEL can be added. CircleCI also checks whether or not it will fail with it.

**Comment by neheb** [comment]
URL can be changed to HTTPS.

**Comment by commodo** [comment]
this does work; it needs some more stuff to do Python3;
from the `configure.ac`:
```
m4_define_default([_AM_PYTHON_INTERPRETER_LIST],[python2 python2.7 python2.6 python2.5 python2.4 python])
AM_PATH_PYTHON([2.4], [HAVE_PYTHON=true], [HAVE_PYTHON=false])
HAVE_PYTHON2=false
if test $HAVE_PYTHON && test "x${PYTHON_VERSION%%.*}" = "x2"; then
    HAVE_PYTHON2=true
fi
```

Also, when this builds, it fails because the shebang is `/usr/bin/python2`
Even if this is fixed, the file needs some adjustment to Python3 format-style.

Also, it seems to put stuff in `/usr/lib/python2.7` [not python3.7].

This does not seem difficult to solve, but I would work with upstream to do this, or backport a patch for Python3 support if they have one already.

**Comment by commodo** [comment]
fixed initial comment;
sorry

**Comment by cotequeiroz** [comment]
> 
> Also, when this builds, it fails because the shebang is /usr/bin/python2
> Even if this is fixed, the file needs some adjustment to Python3 format-style.

Also, it seems to put stuff in /usr/lib/python2.7 [not python3.7].

The source themselves do not have paths hardcoded, but seem to assume that `build-python == (target|host)-python`, so it uses whatever it detectd to create the shebang, or from [PYTHON, if set](https://github.com/nmap/nmap/blob/ec30dbab3f519f91bf6b66653b5c44187cae5a13/configure#L5513), `INSTALL_LIB`, which points to where they are installed, apparently gets set by [ndiff/setup.py](https://github.com/nmap/nmap/blob/ec30dbab3f519f91bf6b66653b5c44187cae5a13/ndiff/setup.py#L101).  The resulting `INSTALL_LIB` is what ends up in `/usr/bin/ndiff`, so just moving `ndiff.py` to `/usr/lib/python3.7` is not enough by itself.

> 
> This does not seem difficult to solve, but I would work with upstream to do this, or backport a patch for Python3 support if they have one already.

nmap/nmap#1484 updates the scripts themselves, apparently only the syntax (run through some 2 to 3 converter?), but it does not take care of the build system (`configure.ac`, for example).  I have not tested it.

**Comment by commodo** [comment]
> [nmap/nmap#1484](https://github.com/nmap/nmap/pull/1484) updates the scripts themselves, apparently only the syntax (run through some 2 to 3 converter?), but it does not take care of the build system (`configure.ac`, for example). I have not tested it.

Looks like a mix of `2to3` tool + manual updates.
A bit better than what I tried locally [just running the 2to3 tool].

**Comment by jefferyto** [comment]
nmap/nmap#1807 is another PR to port ndiff to Python 3. According to https://github.com/nmap/nmap/issues/1176#issuecomment-578320410, Debian has decided to carry the PR as a patch (available [here](https://salsa.debian.org/pkg-security-team/nmap/blob/0510c602dd45f4dc0b06a6f422a9b0855564ddbb/debian/patches/0004-Python3-port-of-ndiff.patch)).

@nunojpg Are you still working on this?

**Comment by nunojpg** [comment]
@jefferyto I am now looking into it.

**Pull request closed without merging**